lib/python/treadmill_aws/ec2client.py

```diff
@@ -86,9 +97,40 @@ def list_instances(ec2_conn, ids=None, tags=None, hostnames=None, state=None):
 
     for reservation in reservations:
         instances.extend(reservation['Instances'])
+
+    # working around a bug in instance-lifetime filter
+    if spot is True:
+        return list(filter(lambda r: 'InstanceLifecycle' in r, instances))
+    elif spot is False:
+        return list(filter(lambda r: 'InstanceLifecycle' not in r, instances))
```

Comment:
Can spot be something else than True/False? Default is spot=None, can we make it a bool?

Reply:
The idea there is 
None (default) = returns spot + on-demand instances
True = returns spot only instances
False = returns on-demand only instances